db/bigtable_eth1.go

```diff
@@ -4593,24 +4579,34 @@ func (bigtable *Bigtable) GetSignature(hex string, st types.SignatureType) (*str
 }
 
 // get a method label for its byte signature with defaults
-func (bigtable *Bigtable) GetMethodLabel(id []byte, invokesContract bool) string {
-	method := "Transfer"
-	if len(id) > 0 {
-		if invokesContract {
-			method = fmt.Sprintf("0x%x", id)
+func (bigtable *Bigtable) GetMethodLabel(data []byte, interaction types.ContractInteractionType) string {
+	id := data
+	if len(data) > 3 {
+		id = data[:4]
+	}
+	method := fmt.Sprintf("0x%x", id)
+
+	switch interaction {
+	case types.CONTRACT_NONE:
+		return "Transfer"
+	case types.CONTRACT_CREATION:
+		return "Constructor"
+	case types.CONTRACT_DESTRUCTION:
+		return "Destruction"
```

Comment:
**nitpick:** **Constructor** and **Destruction** are not the same type of word.
If we want to leave it like that it is also fine.

Reply:
I see what you mean, but I think this is okay because developers can decide how a smart contract is constructed, while that's not possible for destruction. I chose that wording to highlight that difference, but it's not a very strong opinion either.